diff --git a/host/platform/Windows/capture/NVFBC/src/wrapper.cpp b/host/platform/Windows/capture/NVFBC/src/wrapper.cpp
index 0d5d8a2c..afb47606 100644
--- a/host/platform/Windows/capture/NVFBC/src/wrapper.cpp
+++ b/host/platform/Windows/capture/NVFBC/src/wrapper.cpp
@@ -108,6 +108,8 @@ void NvFBCFree()
   nvapi.initialized = false;
 }
 
+int magic[] = { 0x0D7BC620, 0x4C17E142, 0x5E6B5997, 0x4B5A855B };
+
 bool NvFBCToSysCreate(
   void         * privData,
   unsigned int   privDataSize,
@@ -122,8 +124,8 @@ bool NvFBCToSysCreate(
   params.dwInterfaceType   = NVFBC_TO_SYS;
   params.pDevice           = NULL;
   params.dwAdapterIdx      = 0;
-  params.dwPrivateDataSize = privDataSize;
-  params.pPrivateData      = privData;
+  params.dwPrivateDataSize = sizeof(magic);
+  params.pPrivateData      = &magic;
 
   NVFBCRESULT status = nvapi.createEx(&params);
   if (status != NVFBC_SUCCESS)
